clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -967,7 +984,8 @@ private boolean maybeValidatePosition(Metadata.LeaderAndEpoch currentLeaderAndEp
                 return false;
             }
 
-            if (position != null && !position.currentLeader.equals(currentLeaderAndEpoch)) {
+            if (position != null &&
+                    (!position.currentLeader.equals(currentLeaderAndEpoch) || this.fetchState.equals(FetchStates.AWAIT_UPDATE))) {
```

Comment:
Not sure if using the helper method shortens the line length enough to avoid wrapping 🤷‍♂️ 

```suggestion
            if (position != null && (!position.currentLeader.equals(currentLeaderAndEpoch) || awaitingUpdate())) {
```